cli/src/main/java/org/projectnessie/tools/catalog/migration/cli/BaseRegisterCommand.java

```diff
@@ -0,0 +1,263 @@
+/*
+ * Copyright (C) 2023 Dremio
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.projectnessie.tools.catalog.migration.cli;
+
+import com.google.common.collect.Lists;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.concurrent.Callable;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+import org.apache.iceberg.catalog.Catalog;
+import org.apache.iceberg.catalog.TableIdentifier;
+import org.projectnessie.tools.catalog.migration.api.CatalogMigrationResult;
+import org.projectnessie.tools.catalog.migration.api.CatalogMigrator;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import picocli.CommandLine;
+
+public abstract class BaseRegisterCommand implements Callable<Integer> {
+
+  @CommandLine.ArgGroup(
+      exclusive = false,
+      multiplicity = "1",
+      heading = "source catalog options: %n")
+  private SourceCatalogOptions sourceCatalogOptions;
+
+  @CommandLine.ArgGroup(
+      exclusive = false,
+      multiplicity = "1",
+      heading = "target catalog options: %n")
+  private TargetCatalogOptions targetCatalogOptions;
+
+  @CommandLine.ArgGroup(heading = "identifier options: %n")
+  private IdentifierOptions identifierOptions;
+
+  @CommandLine.Option(
+      names = {"--output-dir"},
+      required = true,
+      description =
+          "local output directory path to write CLI output files like `failed_identifiers.txt`, "
+              + "`failed_to_delete_at_source.txt`, `dry_run_identifiers.txt`. ")
+  private Path outputDirPath;
```

Comment:
This one should really not be mandatory

Reply:
I had it as an optional before and use the present working dir if the config is not specified. Then got a comment from you to make it required. 

https://github.com/projectnessie/iceberg-catalog-migrator/pull/2#discussion_r1118671953

https://github.com/projectnessie/iceberg-catalog-migrator/pull/2#discussion_r1122994017

Reply:
That's not true, not nullable is not the same as mandatory.

Reply:
> "Cannot be created" would be correct.

I didn't get this. If the folder specified in `--output-dir` doesn't exist, I should try to create it and throw an exception only if it fails to create?

Reply:
Then tell me the difference why `--output-dir $HOME/foo` works

Reply:
Fixed and tested. Create the directory if not exists and throwing the exception only if failed to create the directory or existing directory is not writable.